### PR TITLE
new TransE framework

### PR DESCRIPTION
**Pull request by menjarleev**
*Issue #, if available:*
* num_worker must be set to 0 to prevent initialization of process overwhelm training time.

*Description of changes:*
* new framework with PyTorch implementation.
* sampling with Pytorch dataloader

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/dglke/models/ke_model.py [hunk 1]
```diff
@@ -976,3 +992,775 @@ def load(self, model_path):
...
+        if len(args.gpu) > 1 and args.num_proc > 1:
+            assert args.num_proc % len(args.gpu) == 0, \
+                'The number of processes needs to be divisible by the number of GPUs'
+
+        if args.neg_deg_sample_eval:
+            assert not args.eval_filter, "if negative sampling based on degree, we can't filter positive edges."
+
+        # need to change to num_nodes if use 0.6.0 dgl version
+        self.initialize(dataset.n_entities, dataset.n_relations, args.init_strat)
+        self.categorize_embedding()
```
Comment (classicsong) [inline python/dglke/models/ke_model.py:1174]: Do not quite understand what categorize_embedding means.

File: python/dglke/models/ke_model.py [hunk 2]
```diff
@@ -976,3 +992,775 @@ def load(self, model_path):
...
+        prepare_save_path(args)
+        if args.save_log:
+            log_file = 'log.txt'
+            result_file = 'result.txt'
+            Logger.log_path = os.path.join(args.save_path, log_file)
+            Logger.result_path = os.path.join(args.save_path, result_file)
+            print = Logger.print
+        init_time_start = time.time()
+        # load dataset and samplers
+        dataset = get_dataset(args.data_path,
```
Comment (classicsong) [inline python/dglke/models/ke_model.py:1234]: Can we put it some where else, just before fit?

File: python/dglke/models/ke_model.py [hunk 3]
```diff
@@ -976,3 +992,775 @@ def load(self, model_path):
...
+                              args.dataset,
+                              args.format,
+                              args.delimiter,
+                              args.data_files,
+                              args.has_edge_importance)
+
+        if args.neg_sample_size_eval < 0:
+            args.neg_sample_size_eval = dataset.n_entities
+        args.batch_size = get_compatible_batch_size(args.batch_size, args.neg_sample_size)
```
Comment (classicsong) [inline python/dglke/models/ke_model.py:1243]: We can put these outside of fit.

**Review by classicsong — COMMENTED** [review]
Let's take sometime to discuss about the interface design and write a RFC about it. 
We need to further refine the whole progress.

**Comment by bamtercelboo** [comment]
When is the code merged into the master？

**Comment by menjarleev** [comment]
> When is the code merged into the master？

It should be done in a month. We are refactoring to a more general framework which supports KGE, GNN and so on.

**Comment by bamtercelboo** [comment]
> 
> 
> > When is the code merged into the master？
> 
> It should be done in a month. We are refactoring to a more general framework which supports KGE, GNN and so on.

Thanks. It's great. I'm looking forward to it.

**Pull request closed without merging**